src/Types/DecimalType.php

```diff
@@ -37,7 +38,7 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
     {
         // Some drivers starting from PHP 8.1 can represent decimals as float/int
         // See also: https://github.com/doctrine/dbal/pull/4818
-        if (PHP_VERSION_ID >= 80100 && (is_float($value) || is_int($value))) {
+        if ((PHP_VERSION_ID >= 80100 || $platform instanceof SqlitePlatform) && (is_float($value) || is_int($value))) {
```

Comment:
The comment is about the driver, and the logic of representing database values in PHP is also about the drivers. Should the condition be based on the driver as well?

Reply:
It turns out that ext-sqlite3 has always behaved like PDO_sqlite started to behave in PHP 8.1 (see #4818 and PHP bug 81397).

imho, we can just drop the PHP version check here, but I guess I wanted to be extra cautious and not open up too many new execution paths on PHP 7. On DBAL 4, we should be good already.

https://github.com/doctrine/dbal/blob/a8a2650e6a0877ab310b346fd294e9a4f888257e/src/Types/DecimalType.php#L27-L33